### PR TITLE
Host.json tracing for "durableTask"

### PR DESCRIPTION
**Pull request by cgillum**
Resolves https://github.com/Azure/azure-functions-host/issues/5422

The [durableTask section of host.json](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-bindings#durable-functions-2-0-host-json) has a large number of settings that users can configure. We want these user-configured settings traced as-is to simplify debugging/supportability.

No secrets or PII are ever expected to be in this section, so it should be safe for tracing.

### CODE REVIEW

**Comment by mhoeger** [comment]
Looking at this again, it looks like the `SanitizeHostJson` logic that uses `WellKnownHostJsonProperties` is only looking one level deep (which works for the host.json of V1 functions but not V2/V3). I'm guessing that sections like "http", "queues", "serviceBus", and "eventHub" aren't being logged properly since they are now nested under "extensions". Not quite as straight-forward as I thought :(

File: src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs [hunk 1]
```diff
@@ -49,7 +49,8 @@ public class HostJsonFileConfigurationProvider : ConfigurationProvider
             private static readonly string[] WellKnownHostJsonProperties = new[]
             {
                 "version", "functionTimeout", "functions", "http", "watchDirectories", "queues", "serviceBus",
-                "eventHub", "singleton", "logging", "aggregator", "healthMonitor", "extensionBundle", "managedDependencies"
+                "eventHub", "singleton", "logging", "aggregator", "healthMonitor", "extensionBundle", "managedDependencies",
```
Comment (mathewc) [inline src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs]: The pattern we use now for logging config options is the IOptionsFormatter mechanism, e.g. https://github.com/Azure/azure-webjobs-sdk/blob/6a6b0d0c3c03282f4be5752fb26f0aef1e154b9c/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Config/QueuesOptions.cs#L17. I think this WellKnownHostJsonProperties thing is obsolete and we should probably remove it. @brettsam ?

Comment (brettsam) [reply src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs]: Yes, this really doesn't apply anymore. It's now up to the Options themselves to tell us what is relevant for logging. This is so we know the "truth" about what options are being used (and allows each extension to tell us what to log in their own code without having to map something here). 

I think this whole thing should go away...

**Comment by fabiocav** [comment]
Closing this as it was originally opened against `dev`, which is now 3.0. Please reopen if still relevant and retarget if needed.

**Pull request closed without merging**